t/dedupe.c

```diff
@@ -304,13 +347,29 @@ static int do_work(struct worker_thread *thread, void *buf)
 	return 0;
 }
 
+static void thread_init_zlib_control(struct worker_thread *thread)
+{
+	z_stream *stream = &thread->zc.stream;
+	stream->zalloc = Z_NULL;
+	stream->zfree = Z_NULL;
+	stream->opaque = Z_NULL;
+
+	if (deflateInit(stream, Z_DEFAULT_COMPRESSION) != Z_OK)
+		return;
+
+	thread->zc.buf_in = fio_memalign(blocksize, blocksize, false);
+	thread->zc.buf_out = fio_memalign(blocksize, deflateBound(stream, blocksize), false);
+}
+
 static void *thread_fn(void *data)
 {
-	struct worker_thread *thread = data;
+	struct worker_thread *thread = my_wt = data;
```

Comment:
Please don't assign the thread variable together with the local one, do that separately.

Reply:
fixed